CONTRIBUTING.md

```diff
@@ -218,6 +218,43 @@ or with the `runtests.jl` script, e.g. to run `test/bitarray.jl` and `test/math.
 
 Make sure that [Travis](http://www.travis-ci.org) greenlights the pull request with a [`Good to merge` message](http://blog.travis-ci.com/2012-09-04-pull-requests-just-got-even-more-awesome).
 
+#### Modifying base more efficiently with Revise.jl
+
+If you install the [Revise](https://github.com/timholy/Revise.jl)
+package, you can make extensive changes to Base without needing to
+rebuild each time.
+
+Here is the basic procedure:
+
+1. If you are planning changes to any types or macros, make those
+   changes, commit them, and build julia using `make`. You do not have
+   to add any functionality based on those types, just the type
+   definitions themselves. (This is necessary because `Revise` cannot
+   handle changes to type definitions or macros.)
+
+2. Start a Julia REPL session. Then issue the following commands:
+
+```julia
+using Revise    # if you aren't launching it in your .juliarc.jl
+Revise.track(Base)
+```
+
+3. Edit files in `base/`, save your edits, and test the
+   functionality. Once you are satisfied that things work as desired,
+   make another commit and rebuild julia.
+
+Should you for some reason need to quit your REPL session and restart
```

Comment:
Is `git stash` required to put the code sources in sync with the precompiled cache? What would happen if `git stash` is forgotten? I would mention something like that to convey that following the exit procedure is really important.